experiments/bforth/bforth.sh

```
@@ -0,0 +1,722 @@
+    read_token token type word
+
+    [[ "$type" == 'w' && "$word" != ';' ]] || die "Expected word name, but got ${token}"
+
+    local word_name="$word"
+
+    while read_token token type word
+    do
+        case "$word" in
+            ';')
+                is_word_defined "${word_name}" && die "Word already defined <$word_name>"
+                udf["${word_name}"]="$word_body"
```

Shellcheck (0.10.0)

[warning] udf appears unused. Verify use (or export if used externally).

(SC2034)